core/src/main/java/org/apache/accumulo/core/client/impl/Namespaces.java

```diff
@@ -100,50 +106,150 @@ private static ZooCache getZooCache(Instance instance) {
     return namespaceMap;
   }
 
-  public static boolean exists(Instance instance, String namespaceId) {
+  public static boolean exists(Instance instance, Namespace.ID namespaceId) {
     ZooCache zc = getZooCache(instance);
     List<String> namespaceIds = zc.getChildren(ZooUtil.getRoot(instance) + Constants.ZNAMESPACES);
-    return namespaceIds.contains(namespaceId);
-  }
-
-  public static String getNamespaceId(Instance instance, String namespace) throws NamespaceNotFoundException {
-    String id = getNameToIdMap(instance).get(namespace);
-    if (id == null)
-      throw new NamespaceNotFoundException(null, namespace, "getNamespaceId() failed to find namespace");
-    return id;
+    return namespaceIds.contains(namespaceId.canonicalID());
   }
 
+  /**
+   * @deprecated Do not use - String for namespace ID is not type safe. Use {@link #getNamespaceName(Instance, Namespace.ID)}
+   */
+  @Deprecated
```

Comment:
I kept the deprecated methods around because of the 2 API methods NamespaceOperations Map<String,String> namespaceIdMap() and TableOperations Map<String,String> tableIdMap().  If I remove the deprecated methods then I either have to make populateMap(Instance, BiConsumer<String,String>) public or create a similar method that returns Map<String,String>.  I figured at least with the methods deprecated, we can communicate the preferred methods.